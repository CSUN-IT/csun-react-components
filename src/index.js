import MetaPrimaryNav from './components/meta/Nav';
import MetaBreadcrumbs from './components/meta/Breadcrumbs';
import MetaSubNav from './components/meta/SubNav';
import MetaButton from './components/meta/Button';
import MetaContainer from './components/meta/Container';
import MetaRow from './components/meta/Row';
import MetaColumn from './components/meta/Column';
import MetaFormField from './components/meta/FormField';
import MetaFormGroup from './components/meta/FormGroup';
import MetaTag from './components/meta/Tag';
import MetaPanel from './components/meta/Panel';
import MetaAlert from './components/meta/Alert';
import MetaList from './components/meta/List';
import MetaSideNav from './components/meta/SideNav';
import MetaAccordion from './components/meta/Accordion';
import MetaChatBubble from './components/web/ChatBubble';
import WebPrimaryNav from './components/web/PrimaryNav';
import WebHeader from './components/web/Header';
import WebSidebar from './components/web/Sidebar';

export {
  MetaBreadcrumbs,
  MetaPrimaryNav,
  MetaSubNav,
  MetaButton,
  MetaContainer,
  MetaRow,
  MetaColumn,
  MetaTag,
  MetaPanel,
  MetaFormField,
  MetaFormGroup,
  MetaAlert,
  MetaList,
  MetaSideNav,
  MetaAccordion,
  MetaChatBubble,
  WebPrimaryNav,
  WebHeader,
  WebSidebar,
};
