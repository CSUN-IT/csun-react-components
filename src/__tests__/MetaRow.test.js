import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { MetaColumn, MetaRow } from '../../src';

configure({ adapter: new Adapter() });

describe('MetaRow', () => {
  it('Should be defined', () => {
    expect(MetaRow).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <MetaRow>
        <MetaColumn sizes={['sm']} values={['4']}>
          1
        </MetaColumn>
      </MetaRow>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders children correctly', () => {
    const tree = mount(
      <MetaRow>
        <MetaColumn sizes={['sm']} values={['4']}>
          1
        </MetaColumn>
        <MetaColumn sizes={['sm']} values={['4']}>
          2
        </MetaColumn>
      </MetaRow>,
    );
    expect(tree.find(MetaColumn)).toHaveLength(2);
  });
});
