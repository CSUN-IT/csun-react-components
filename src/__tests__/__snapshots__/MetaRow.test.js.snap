// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaRow Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaRow
    className=""
  >
    <MetaColumn
      className=""
      sizes={
        Array [
          "sm",
        ]
      }
      values={
        Array [
          "4",
        ]
      }
    >
      1
    </MetaColumn>
  </MetaRow>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MetaColumn
        className=""
        sizes={
          Array [
            "sm",
          ]
        }
        values={
          Array [
            "4",
          ]
        }
      >
        1
      </MetaColumn>,
      "className": " row ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "1",
        "className": "",
        "sizes": Array [
          "sm",
        ],
        "values": Array [
          "4",
        ],
      },
      "ref": null,
      "rendered": "1",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MetaColumn
          className=""
          sizes={
            Array [
              "sm",
            ]
          }
          values={
            Array [
              "4",
            ]
          }
        >
          1
        </MetaColumn>,
        "className": " row ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "1",
          "className": "",
          "sizes": Array [
            "sm",
          ],
          "values": Array [
            "4",
          ],
        },
        "ref": null,
        "rendered": "1",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
